src/constants/transformation/transform-service.ts

```
@@ -13,58 +19,215 @@ export class TransformService {
       const tenant = data.tenantData?.[0] ?? {};
 
-      // Extract custom field values
-      const extractField = (label: string) =>
-        data.customFields?.find((f: any) => f.label === label)
-          ?.selectedValues?.[0]?.value ?? null;
+      // Extract custom field values from the new Kafka message structure
+      const extractCustomField = (label: string) => {
+        if (!data.customFields || !Array.isArray(data.customFields)) {
+          return null;
+        }
+        
```

ESLint

[error] Delete `········`

(prettier/prettier)